### PR TITLE
batcheval: Disable use of time-bound iterators in KV commands

### PR DESCRIPTION
**Pull request by bdarnell**
(first commit is #32311)

This optimization has not been shown to make a difference in
benchmarks, and there are doubts about its correctness (#28358). It is
still used for CDC (where it does make a difference) and incremental
backups.

### CODE REVIEW

**Comment by tbg** [comment]
LGTM

File: pkg/ccl/storageccl/engineccl/mvcc_test.go [hunk 1]
```diff
@@ -388,7 +388,7 @@ func TestMVCCIncrementalIteratorIntentStraddlesSStables(t *testing.T) {
 	ingest := func(it engine.Iterator, count int) {
 		sst, err := engine.MakeRocksDBSstFileWriter()
 		if err != nil {
-			t.Fatal(sst)
+			t.Fatal(err)
```
Comment (benesch) [inline pkg/ccl/storageccl/engineccl/mvcc_test.go]: Whoops.

**Review by benesch — APPROVED** [review]

**Comment by bdarnell** [comment]
bors r=benesch

**Review by petermattis — COMMENTED** [review]
:lgtm: This makes me feel much better than the hack.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/32313#-:-LRJ5Yak83sGZK4lrbQt:b-rxmk6c)* status: :shipit: complete! 1 of 0 LGTMs obtained

**Pull request merged**